lib/services/elasticsearch.js

```diff
@@ -581,12 +593,15 @@ class ElasticSearch extends Service {
       }
     };
 
-    if (!this.kuzzle.indexCache.exists(esRequest.index)) {
-      return Bluebird.reject(new PreconditionError(`Index '${esRequest.index}' doesn't exist`));
-    }
+    return this.kuzzle.indexCache.exists(esRequest.index)
+      .then(exists => {
+        if (! exists) {
+          throw new PreconditionError(`Index '${esRequest.index}' and/or collection '${esRequest.type}' does not exist`);
```

Comment:
According to this message, we're telling clients that their `collection:create` action failed, and that's may be because the collection they want to create does not exist :open_mouth:

Reply:
indeed, they may be a little confused